app/models/miq_queue.rb

```diff
@@ -545,6 +545,14 @@ def check_for_timeout(log_prefix = "MIQ(MiqQueue.check_for_timeout)", grace = 10
     end
   end
 
+  def self.candidates_for_timeout
+    where(:state => STATE_DEQUEUE).where("(select date_part('epoch', updated_on) + msg_timeout) < ?", Time.now.to_i)
+  end
```

Comment:
Since this is now a dedicated method can we get unit tests for it?

Reply:
Additionally can we get tests for different msg_timeout values?

Reply:
Done